src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -715,8 +705,6 @@ - (void)_handleSubscriptionReset
         [self _reattemptSubscriptionNowIfNeeded];
         os_unfair_lock_unlock(&self->_lock);
```

Comment:
Could do a lock_guard in this block too.

Reply:
Tried that, it's more lines of code as you need to pass in weakSelf, etc.